src/app/beer_garden/command_publishing_blocklist.py

```diff
@@ -35,80 +38,117 @@ def command_publishing_blocklist_add(command: dict):
     Args:
         blocked_command: a CommandPublishingBlocklist object
+        blocked_command_id: a string of an id for CommandPublishingBlocklist
     """
+
+    if blocked_command is None:
+        blocked_command = CommandPublishingBlocklist.objects.get(id=blocked_command_id)
+
     if config.get("garden.name") != blocked_command.namespace:
         import beer_garden.router
+        from beer_garden.api.http import CommandPublishingBlocklistSchema
 
+        target_garden = Garden.objects.get(
+            namespaces__contains=blocked_command["namespace"]
+        ).name
         beer_garden.router.route(
             Operation(
                 operation_type="COMMAND_BLOCKLIST_REMOVE",
-                args=[blocked_command.id],
-                target_garden_name=blocked_command.namespace,
+                kwargs={"blocked_command_id": f"{blocked_command.id}"},
+                target_garden_name=target_garden,
             )
         )
         blocked_command.status = "REMOVE_REQUESTED"
         blocked_command.save()
+        publish(
+            Event(
+                garden=config.get("garden.name"),
+                name=Events.COMMAND_PUBLISHING_BLOCKLIST_UPDATE.name,
+                metadata={
+                    "blocked_command": CommandPublishingBlocklistSchema()
+                    .dump(blocked_command)
+                    .data
+                },
+            )
+        )
     else:
+        publish(
+            Event(
+                garden=config.get("garden.name"),
+                name=Events.COMMAND_PUBLISHING_BLOCKLIST_REMOVE.name,
+                metadata={"id": f"{blocked_command.id}"},
+            )
+        )
         blocked_command.delete()
 
 
-def command_publishing_blocklist_remove(command_publishing_id: str):
-    """Deletes the CommandPublishingBlocklist object with corresponding id from database.
+def _update_blocklist(blocked_command):
+    try:
+        local_blocked_command = CommandPublishingBlocklist.objects.get(
+            namespace=blocked_command["namespace"],
+            system=blocked_command["system"],
+            command=blocked_command["command"],
+        )
 
-    Args:
-        command_publishing_id: a string of an id used to get CommandPublishingBlocklist object from database
-    """
-    CommandPublishingBlocklist.objects.filter(id=command_publishing_id).delete()
+        if f"{local_blocked_command.id}" != blocked_command["id"]:
+            local_blocked_command.delete()
+            local_blocked_command = CommandPublishingBlocklist(**blocked_command)
+        else:
+            local_blocked_command.status = blocked_command["status"]
+        local_blocked_command.save()
+    except DoesNotExist:
+        CommandPublishingBlocklist(**blocked_command).save()
+
+
+def _handle_update_event(event):
+    _update_blocklist(event.metadata["blocked_command"])
+
+
+def _handle_remove_event(event):
+    CommandPublishingBlocklist.objects.filter(id=event.metadata["id"]).delete()
```

Comment:
I suppose I should know the answer to this, but does another delete event need to be generated and published for this?  I'm specifically thinking about the grandchild scenario.  If an entry is deleted directly on the grandchild, would the published event reach the top level parent, or does the child need to publish the delete event?

As an aside, I tried to test this out for myself, but I still couldn't add a command for a grandchild garden.  Is that expected to work?

Reply:
When I delete a command from the grandchild it is getting removed from the parent. Adding a command to the grandchild should be working because I am able to remove and add commands to a grandchild from the parent.